app/Jobs/SendToBlink.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace Rogue\Jobs;
+
+use Illuminate\Bus\Queueable;
+use DoSomething\Gateway\Blink;
+use Illuminate\Queue\SerializesModels;
+use Illuminate\Queue\InteractsWithQueue;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Foundation\Bus\Dispatchable;
+
+class SendToBlink implements ShouldQueue
+{
+    use Dispatchable, InteractsWithQueue, Queueable, SerializesModels;
+
+    /**
+     * Blink API client.
+     *
+     * @var \DoSomething\Gateway\Blink
+     */
+    protected $blink;
+
+    /**
+     * Create a new job instance.
+     *
+     * @return void
+     */
+    public function __construct(Blink $blink)
+    {
+        $this->blink = $blink;
+    }
+
+    /**
+     * Execute the job.
+     *
+     * @return void
+     */
+    public function handle($payload)
+    {
+        // If there is a signup_id, this is a userSignupPost.
+        // Otherwise, it is a userSignup.
+        if ($payload['signup_id']) {
```

Comment:
Can we make two separate queue jobs for this? (One `SendSignupToBlink` and one `SendPostToBlink`, with the `Signup $signup` or `Post $post` as the job's argument?)

Reply:
yes that makes more sense!